packages/wrangler/src/__tests__/pages/publish.test.ts

```diff
@@ -2178,6 +2178,413 @@ and that at least one include rule is provided.
+				const body = (await req.json()) as {
+					hashes: string[];
+				};
+
+				expect(req.headers.get("Authorization")).toBe(
+					"Bearer <<funfetti-auth-jwt>>"
+				);
+				expect(body).toMatchObject({
+					hashes: ["13a03eaf24ae98378acd36ea00f77f2f"],
+				});
+
+				return res.once(
+					ctx.status(200),
+					ctx.json({
+						success: true,
+						errors: [],
+						messages: [],
+						result: body.hashes,
+					})
+				);
+			}),
+
+			// /pages/assets/upload
+			rest.post("*/pages/assets/upload", async (req, res, ctx) => {
+				expect(req.headers.get("Authorization")).toBe(
+					"Bearer <<funfetti-auth-jwt>>"
+				);
+
+				expect(await req.json()).toMatchObject([
+					{
+						key: "13a03eaf24ae98378acd36ea00f77f2f",
+						value: Buffer.from("This is a readme").toString("base64"),
+						metadata: {
+							contentType: "text/markdown",
+						},
+						base64: true,
+					},
+				]);
+
+				return res.once(
+					ctx.status(200),
+					ctx.json({
+						success: true,
+						errors: [],
+						messages: [],
+						result: null,
+					})
+				);
+			}),
+
+			// /accounts/:accountId/pages/projects/<project-name>/deployments
+			rest.post(
+				"*/accounts/:accountId/pages/projects/foo/deployments",
+				async (req, res, ctx) => {
+					const body = await (req as RestRequestWithFormData).formData();
+					const manifest = JSON.parse(body.get("manifest") as string);
+					const workerBundle = body.get("_worker.bundle") as string;
+
+					expect(req.params.accountId).toEqual("some-account-id");
+					// make sure this is all we uploaded
+					expect([...body.keys()].sort()).toEqual(
+						[
+							"manifest",
+							"_worker.bundle",
+							"functions-filepath-routing-config.json",
+							"_routes.json",
+						].sort()
+					);
+					expect(manifest).toMatchInlineSnapshot(`
+				Object {
+				  "/README.md": "13a03eaf24ae98378acd36ea00f77f2f",
+				}
+			`);
+
+					// some fields in workerBundle, such as the undici form boundary
+					// or the file hashes, are randomly generated. Let's replace these
+					// dynamic values with static ones so we can properly test the
+					// contents of `workerBundle`
+					// see https://jestjs.io/docs/snapshot-testing#property-matchers
+					let workerBundleWithConstantData = workerBundle.replace(
+						/------formdata-undici-0.[0-9]*/g,
+						"------formdata-undici-0.test"
+					);
+					workerBundleWithConstantData = workerBundleWithConstantData.replace(
+						/functionsWorker-0.[0-9]*.js/g,
+						"functionsWorker-0.test.js"
+					);
+					workerBundleWithConstantData = workerBundleWithConstantData.replace(
+						/[0-9a-z]*-hello.wasm/g,
+						"test-hello.wasm"
+					);
+
+					// check we appended the metadata
+					expect(workerBundleWithConstantData).toContain(
+						`Content-Disposition: form-data; name="metadata"`
+					);
+					expect(workerBundleWithConstantData).toContain(
+						`{"main_module":"functionsWorker-0.test.js","bindings":[]}`
```

Comment:
Likewise, it would be nice if we could omit the empty `"bindings":[]`, but not critical and we can do it in cleanup if we still care enough.